src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceCollectionDescriptorExtensionsTests.cs

```diff
@@ -2,15 +2,54 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
 using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
+using Microsoft.Extensions.DependencyInjection.Tests;
 using Xunit;
-using Xunit.Abstractions;
 
 namespace Microsoft.Extensions.DependencyInjection
 {
+    public class ABC { }
+
     public class ServiceCollectionDescriptorExtensionsTest
     {
+        [Fact]
+        internal void GetService_FactoryCallSite_Transient_DoesNotFail()
+        {
+            TestEventListener listener = new TestEventListener();
```

Comment:
You know this might not be a bad idea to enable generally for all of these tests.

Reply:
There is already an Debug.Fail in that branch.https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/DynamicServiceProviderEngine.cs#L42

Reply:
I removed the event listener and would just rely on the Debug.Fail